src/Components/Web.JS/src/Boot.Server.Common.ts

```diff
@@ -0,0 +1,201 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { DotNet } from '@microsoft/dotnet-js-interop';
+import { Blazor } from './GlobalExports';
+import { HubConnectionBuilder, HubConnection, HttpTransportType } from '@microsoft/signalr';
+import { MessagePackHubProtocol } from '@microsoft/signalr-protocol-msgpack';
+import { showErrorNotification } from './BootErrors';
+import { RenderQueue } from './Platform/Circuits/RenderQueue';
+import { ConsoleLogger } from './Platform/Logging/Loggers';
+import { LogLevel, Logger } from './Platform/Logging/Logger';
+import { CircuitDescriptor } from './Platform/Circuits/CircuitManager';
+import { resolveOptions, CircuitStartOptions } from './Platform/Circuits/CircuitStartOptions';
+import { DefaultReconnectionHandler } from './Platform/Circuits/DefaultReconnectionHandler';
+import { attachRootComponentToLogicalElement } from './Rendering/Renderer';
+import { discoverComponents, discoverPersistedState, ServerComponentDescriptor } from './Services/ComponentDescriptorDiscovery';
+import { sendJSDataStream } from './Platform/Circuits/CircuitStreamingInterop';
+import { fetchAndInvokeInitializers } from './JSInitializers/JSInitializers.Server';
+
+let renderingFailed = false;
+let connection: HubConnection;
+let circuit: CircuitDescriptor;
+let dispatcher: DotNet.ICallDispatcher;
```

Comment:
Before this PR, we had `DotNet.jsCallDispatcher` to dispatch into JS, and `DotNet.invokeMethod{Async}` to call static .NET methods. I've introduced a new `ICallDispatcher` to replace both of these.

Each `ICallDispatcher` is associated with a `DotNetCallDispatcher`, which each platform provides an implementation of (this part hasn't changed). When a .NET->JS call occurs, we then know which .NET runtime to "return" to, based on the `ICallDispatcher` that initiated the JS function call. `DotNetObject` has been extended to include a reference to its `ICallDispatcher` so it knows which .NET runtime to invoke into.